controllers/k8sgpt_controller.go

```diff
@@ -209,6 +209,12 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		// Parse the k8sgpt-deployment response into a list of results
 		k8sgptNumberOfResults.Set(float64(len(response.Results)))
 		rawResults := make(map[string]corev1alpha1.Result)
+		integration, err := integrations.NewIntegrations(r.Client, ctx)
```

Comment:
I would probably create this in the main and pass it through as a field on the k8sgpt_controller struct

Reply:
not sure how this will work without refactoring Integrations, can you share a pseudo code example?

Reply:
```
	if err = (&controllers.K8sGPTReconciler{
		Client: mgr.GetClient(),
		Scheme: mgr.GetScheme(),
                integrations.NewIntegrations(mgr.GetClient(), ctx)
	}).SetupWithManager(mgr); err != nil {
		setupLog.Error(err, "unable to create controller", "controller", "K8sGPT")
		os.Exit(1)
	}
```

Reply:
This way, we are not recreating the `NewIntegrations` on each reconcile

Reply:
cool, I misread your message earlier today.
Thanks for all the recommendations and patience,  I've a better grasp of the project.